src/signing_key.rs

```diff
@@ -106,6 +115,65 @@ impl From<[u8; 32]> for SigningKey {
     }
 }
 
+impl<'a> TryFrom<PrivateKeyInfo<'a>> for SigningKey {
+    type Error = Error;
+    fn try_from(pki: PrivateKeyInfo) -> Result<Self, Error> {
+        if pki.algorithm == ALGORITHM_ID {
+            SigningKey::try_from(pki.private_key)
+        } else {
+            Err(Error::MalformedSecretKey)
+        }
+    }
+}
+
+impl ToPrivateKey for SigningKey {
+    fn to_pkcs8_der(&self) -> Result<PrivateKeyDocument, pkcs8::Error> {
```

Comment:
You can also write these `Result`s as:

```suggestion
    fn to_pkcs8_der(&self) -> pkcs8::Result<PrivateKeyDocument> {
```